### PR TITLE
layers: Don't add VK_SUBPASS_EXTERNAL edges to subpass DAG.

### LINKED ISSUE #978: Crash when dependency.srcSubpass is VK_SUBPASS_EXTERNAL
Call stack:

`VkLayer_core_validation.dll!core_validation::CheckPreserved(const core_validation::layer_data * my_data, const VkRenderPassCreateInfo * pCreateInfo, const int index, const unsigned int attachment, const std::vector<DAGNode,std::allocator<DAGNode> > & subpass_to_node, int depth, bool & skip_call) Line 9468	C++`

`VkLayer_core_validation.dll!core_validation::CheckPreserved(const core_validation::layer_data * my_data, const VkRenderPassCreateInfo * pCreateInfo, const int index, const unsigned int attachment, const std::vector<DAGNode,std::allocator<DAGNode> > & subpass_to_node, int depth, bool & skip_call) Line 9482	C++`

`VkLayer_core_validation.dll!core_validation::ValidateDependencies(const core_validation::layer_data * my_data, const FRAMEBUFFER_NODE * framebuffer, const RENDER_PASS_NODE * renderPass) Line 9646	C++`

`VkLayer_core_validation.dll!core_validation::CmdBeginRenderPass(VkCommandBuffer_T * commandBuffer, const VkRenderPassBeginInfo * pRenderPassBegin, VkSubpassContents contents) Line 10303	C++`

`VkLayer_object_tracker.dll!object_tracker::CmdBeginRenderPass(VkCommandBuffer_T * commandBuffer, const VkRenderPassBeginInfo * pRenderPassBegin, VkSubpassContents contents) Line 2514	C++
VkLayer_parameter_validation.dll!parameter_validation::CmdBeginRenderPass(VkCommandBuffer_T * commandBuffer, const VkRenderPassBeginInfo * pRenderPassBegin, VkSubpassContents contents) Line 4643	C++
`

`VkLayer_threading.dll!threading::CmdBeginRenderPass(VkCommandBuffer_T * commandBuffer, const VkRenderPassBeginInfo * pRenderPassBegin, VkSubpassContents contents) Line 2980	C++`

`71_Deferred.exe!Render(float alpha, float elapsedtime) Line 1107	C++`

`71_Deferred.exe!main(int argc, char * * argv) Line 815	C++`

### PR DESCRIPTION
**Pull request by chrisforbes**
We don't use them, and handling them safely complicates other code that
works with the DAG.

Will fix #978.

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -9791,23 +9791,28 @@ static bool CreatePassDAG(const layer_data *my_data, VkDevice device, const VkRe
...
+                skip_call |= log_msg(my_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, (VkDebugReportObjectTypeEXT)0, 0, __LINE__,
+                                     DRAWSTATE_INVALID_RENDERPASS, "DS", "The src and dest subpasses cannot both be external.");
+            }
+
+            // We don't want to add edges to the DAG for dependencies to/from
+            // VK_SUBPASS_EXTERNAL. We don't use them for anything, and their
+            // presence complicates other code.
+            continue;
+        }
+        else if (dependency.srcSubpass > dependency.dstSubpass) {
```
Comment (tobine) [inline layers/core_validation.cpp]: clang-format? I believe these "else if" blocks should get rolled onto the same line as previous closing "}"

**Comment by chrisforbes** [comment]
Style fixed, running on the farm now.

**Comment by chrisforbes** [comment]
Is clean on the farm

**Comment by chrisforbes** [comment]
Pushed to master.

**Pull request closed without merging**